### ISSUE TITLE
Codec Error Shortly After Video Prepared

### ISSUE THREAD

**Issue by findyourexit** [google/ExoPlayer#9879]
I've run into an issue with playback of video content on some handsets and have spent a bunch of time isolating the issue in an attempt to ascertain the root cause (I feel like I got close!). 

The issue occurs moments (approximately 3 seconds) after preparing a particular media file (`.mp4` format), if the user does not commence video playback almost immediately. During that window of approximately 3 seconds, playback of the video occurs without an issue, however if the user hesitates to hit the "play" button to initiate playback within that window of approximately 3 seconds, the codec seems to throw an error (included trace, below).

```
2022-01-20 13:07:48.869 10159-10224/com.google.android.exoplayer2.demo D/BufferPoolAccessor2.0: bufferpool2 0x7388e36088 : 5(40960 size) total buffers - 0(0 size) used buffers - 19/24 (recycle/alloc) - 5/24 (fetch/transfer)
2022-01-20 13:07:48.869 10159-10224/com.google.android.exoplayer2.demo D/BufferPoolAccessor2.0: evictor expired: 1, evicted: 1
2022-01-20 13:07:49.444 10159-10222/com.google.android.exoplayer2.demo W/CCodec: [c2.qti.avc.decoder] previous call to queue exceeded timeout
2022-01-20 13:07:49.445 10159-10220/com.google.android.exoplayer2.demo E/MediaCodec: Codec reported err 0x80000000, actionCode 0, while in state 6/STARTED
2022-01-20 13:07:49.445 10159-10220/com.google.android.exoplayer2.demo D/SurfaceUtils: disconnecting from surface 0x7428e222b0, reason disconnectFromSurface
2022-01-20 13:07:49.447 10159-10220/com.google.android.exoplayer2.demo D/CCodecBufferChannel: [c2.qti.avc.decoder#469] MediaCodec discarded an unknown buffer
2022-01-20 13:07:49.447 10159-10220/com.google.android.exoplayer2.demo E/BufferQueueProducer: [SurfaceView[com.google.android.exoplayer2.demo/com.google.android.exoplayer2.demo.PlayerActivity]#2(BLAST Consumer)2](id:27af00000002,api:0,p:-1,c:10159) cancelBuffer: BufferQueue has no connected producer
2022-01-20 13:07:49.447 10159-10220/com.google.android.exoplayer2.demo D/CCodecBufferChannel: [c2.qti.avc.decoder#469] MediaCodec discarded an unknown buffer
2022-01-20 13:07:49.447 10159-10220/com.google.android.exoplayer2.demo E/BufferQueueProducer: [SurfaceView[com.google.android.exoplayer2.demo/com.google.android.exoplayer2.demo.PlayerActivity]#2(BLAST Consumer)2](id:27af00000002,api:0,p:-1,c:10159) cancelBuffer: BufferQueue has no connected producer
2022-01-20 13:07:49.448 10159-10220/com.google.android.exoplayer2.demo D/CCodecBufferChannel: [c2.qti.avc.decoder#469] MediaCodec discarded an unknown buffer
2022-01-20 13:07:49.448 10159-10220/com.google.android.exoplayer2.demo E/BufferQueueProducer: [SurfaceView[com.google.android.exoplayer2.demo/com.google.android.exoplayer2.demo.PlayerActivity]#2(BLAST Consumer)2](id:27af00000002,api:0,p:-1,c:10159) cancelBuffer: BufferQueue has no connected producer
2022-01-20 13:07:49.448 10159-10220/com.google.android.exoplayer2.demo D/CCodecBufferChannel: [c2.qti.avc.decoder#469] MediaCodec discarded an unknown buffer
2022-01-20 13:07:49.448 10159-10220/com.google.android.exoplayer2.demo E/BufferQueueProducer: [SurfaceView[com.google.android.exoplayer2.demo/com.google.android.exoplayer2.demo.PlayerActivity]#2(BLAST Consumer)2](id:27af00000002,api:0,p:-1,c:10159) cancelBuffer: BufferQueue has no connected producer
2022-01-20 13:07:49.449 10159-10220/com.google.android.exoplayer2.demo D/CCodecBufferChannel: [c2.qti.avc.decoder#469] MediaCodec discarded an unknown buffer
2022-01-20 13:07:49.449 10159-10220/com.google.android.exoplayer2.demo E/BufferQueueProducer: [SurfaceView[com.google.android.exoplayer2.demo/com.google.android.exoplayer2.demo.PlayerActivity]#2(BLAST Consumer)2](id:27af00000002,api:0,p:-1,c:10159) cancelBuffer: BufferQueue has no connected producer
2022-01-20 13:07:49.449 10159-10220/com.google.android.exoplayer2.demo D/CCodecBufferChannel: [c2.qti.avc.decoder#469] MediaCodec discarded an unknown buffer
2022-01-20 13:07:49.449 10159-10220/com.google.android.exoplayer2.demo E/BufferQueueProducer: [SurfaceView[com.google.android.exoplayer2.demo/com.google.android.exoplayer2.demo.PlayerActivity]#2(BLAST Consumer)2](id:27af00000002,api:0,p:-1,c:10159) cancelBuffer: BufferQueue has no connected producer
2022-01-20 13:07:49.449 10159-10220/com.google.android.exoplayer2.demo D/CCodecBufferChannel: [c2.qti.avc.decoder#469] MediaCodec discarded an unknown buffer
2022-01-20 13:07:49.449 10159-10220/com.google.android.exoplayer2.demo E/BufferQueueProducer: [SurfaceView[com.google.android.exoplayer2.demo/com.google.android.exoplayer2.demo.PlayerActivity]#2(BLAST Consumer)2](id:27af00000002,api:0,p:-1,c:10159) cancelBuffer: BufferQueue has no connected producer
2022-01-20 13:07:49.450 10159-10220/com.google.android.exoplayer2.demo D/CCodecBufferChannel: [c2.qti.avc.decoder#469] MediaCodec discarded an unknown buffer
2022-01-20 13:07:49.450 10159-10220/com.google.android.exoplayer2.demo E/BufferQueueProducer: [SurfaceView[com.google.android.exoplayer2.demo/com.google.android.exoplayer2.demo.PlayerActivity]#2(BLAST Consumer)2](id:27af00000002,api:0,p:-1,c:10159) cancelBuffer: BufferQueue has no connected producer
2022-01-20 13:07:49.450 10159-10220/com.google.android.exoplayer2.demo D/CCodecBufferChannel: [c2.qti.avc.decoder#469] MediaCodec discarded an unknown buffer
2022-01-20 13:07:49.450 10159-10220/com.google.android.exoplayer2.demo E/BufferQueueProducer: [SurfaceView[com.google.android.exoplayer2.demo/com.google.android.exoplayer2.demo.PlayerActivity]#2(BLAST Consumer)2](id:27af00000002,api:0,p:-1,c:10159) cancelBuffer: BufferQueue has no connected producer
2022-01-20 13:07:49.458 10159-10225/com.google.android.exoplayer2.demo E/BufferQueueProducer: [SurfaceView[com.google.android.exoplayer2.demo/com.google.android.exoplayer2.demo.PlayerActivity]#2(BLAST Consumer)2](id:27af00000002,api:0,p:-1,c:10159) cancelBuffer: BufferQueue has no connected producer
2022-01-20 13:07:49.459 10159-10225/com.google.android.exoplayer2.demo E/BufferQueueProducer: [SurfaceView[com.google.android.exoplayer2.demo/com.google.android.exoplayer2.demo.PlayerActivity]#2(BLAST Consumer)2](id:27af00000002,api:0,p:-1,c:10159) cancelBuffer: BufferQueue has no connected producer
2022-01-20 13:07:49.460 10159-10225/com.google.android.exoplayer2.demo E/BufferQueueProducer: [SurfaceView[com.google.android.exoplayer2.demo/com.google.android.exoplayer2.demo.PlayerActivity]#2(BLAST Consumer)2](id:27af00000002,api:0,p:-1,c:10159) cancelBuffer: BufferQueue has no connected producer
2022-01-20 13:07:49.462 10159-10225/com.google.android.exoplayer2.demo E/BufferQueueProducer: [SurfaceView[com.google.android.exoplayer2.demo/com.google.android.exoplayer2.demo.PlayerActivity]#2(BLAST Consumer)2](id:27af00000002,api:0,p:-1,c:10159) cancelBuffer: BufferQueue has no connected producer
2022-01-20 13:07:49.463 10159-10225/com.google.android.exoplayer2.demo E/BufferQueueProducer: [SurfaceView[com.google.android.exoplayer2.demo/com.google.android.exoplayer2.demo.PlayerActivity]#2(BLAST Consumer)2](id:27af00000002,api:0,p:-1,c:10159) cancelBuffer: BufferQueue has no connected producer
2022-01-20 13:07:49.465 10159-10225/com.google.android.exoplayer2.demo E/BufferQueueProducer: [SurfaceView[com.google.android.exoplayer2.demo/com.google.android.exoplayer2.demo.PlayerActivity]#2(BLAST Consumer)2](id:27af00000002,api:0,p:-1,c:10159) cancelBuffer: BufferQueue has no connected producer
2022-01-20 13:07:49.467 10159-10225/com.google.android.exoplayer2.demo E/BufferQueueProducer: [SurfaceView[com.google.android.exoplayer2.demo/com.google.android.exoplayer2.demo.PlayerActivity]#2(BLAST Consumer)2](id:27af00000002,api:0,p:-1,c:10159) cancelBuffer: BufferQueue has no connected producer
2022-01-20 13:07:49.469 10159-10225/com.google.android.exoplayer2.demo E/BufferQueueProducer: [SurfaceView[com.google.android.exoplayer2.demo/com.google.android.exoplayer2.demo.PlayerActivity]#2(BLAST Consumer)2](id:27af00000002,api:0,p:-1,c:10159) cancelBuffer: BufferQueue has no connected producer
2022-01-20 13:07:49.471 10159-10225/com.google.android.exoplayer2.demo E/BufferQueueProducer: [SurfaceView[com.google.android.exoplayer2.demo/com.google.android.exoplayer2.demo.PlayerActivity]#2(BLAST Consumer)2](id:27af00000002,api:0,p:-1,c:10159) cancelBuffer: BufferQueue has no connected producer
2022-01-20 13:07:49.473 10159-10225/com.google.android.exoplayer2.demo E/BufferQueueProducer: [SurfaceView[com.google.android.exoplayer2.demo/com.google.android.exoplayer2.demo.PlayerActivity]#2(BLAST Consumer)2](id:27af00000002,api:0,p:-1,c:10159) cancelBuffer: BufferQueue has no connected producer
2022-01-20 13:07:49.475 10159-10225/com.google.android.exoplayer2.demo E/BufferQueueProducer: [SurfaceView[com.google.android.exoplayer2.demo/com.google.android.exoplayer2.demo.PlayerActivity]#2(BLAST Consumer)2](id:27af00000002,api:0,p:-1,c:10159) cancelBuffer: BufferQueue has no connected producer
2022-01-20 13:07:49.476 10159-10225/com.google.android.exoplayer2.demo E/BufferQueueProducer: [SurfaceView[com.google.android.exoplayer2.demo/com.google.android.exoplayer2.demo.PlayerActivity]#2(BLAST Consumer)2](id:27af00000002,api:0,p:-1,c:10159) cancelBuffer: BufferQueue has no connected producer
2022-01-20 13:07:49.478 10159-10225/com.google.android.exoplayer2.demo E/BufferQueueProducer: [SurfaceView[com.google.android.exoplayer2.demo/com.google.android.exoplayer2.demo.PlayerActivity]#2(BLAST Consumer)2](id:27af00000002,api:0,p:-1,c:10159) cancelBuffer: BufferQueue has no connected producer
2022-01-20 13:07:49.480 10159-10225/com.google.android.exoplayer2.demo E/BufferQueueProducer: [SurfaceView[com.google.android.exoplayer2.demo/com.google.android.exoplayer2.demo.PlayerActivity]#2(BLAST Consumer)2](id:27af00000002,api:0,p:-1,c:10159) cancelBuffer: BufferQueue has no connected producer
2022-01-20 13:07:49.482 10159-10225/com.google.android.exoplayer2.demo E/BufferQueueProducer: [SurfaceView[com.google.android.exoplayer2.demo/com.google.android.exoplayer2.demo.PlayerActivity]#2(BLAST Consumer)2](id:27af00000002,api:0,p:-1,c:10159) cancelBuffer: BufferQueue has no connected producer
2022-01-20 13:07:49.483 10159-10225/com.google.android.exoplayer2.demo E/BufferQueueProducer: [SurfaceView[com.google.android.exoplayer2.demo/com.google.android.exoplayer2.demo.PlayerActivity]#2(BLAST Consumer)2](id:27af00000002,api:0,p:-1,c:10159) cancelBuffer: BufferQueue has no connected producer
2022-01-20 13:07:49.495 10159-10221/com.google.android.exoplayer2.demo I/hw-BpHwBinder: onLastStrongRef automatically unlinking death recipients
2022-01-20 13:07:49.495 10159-10221/com.google.android.exoplayer2.demo D/BufferPoolAccessor2.0: bufferpool2 0x7388e36e68 : 4(12441600 size) total buffers - 4(12441600 size) used buffers - 27/32 (recycle/alloc) - 6/32 (fetch/transfer)
2022-01-20 13:07:49.495 10159-10225/com.google.android.exoplayer2.demo E/BufferQueueProducer: [SurfaceView[com.google.android.exoplayer2.demo/com.google.android.exoplayer2.demo.PlayerActivity]#2(BLAST Consumer)2](id:27af00000002,api:0,p:-1,c:10159) detachBuffer: BufferQueue has no connected producer
2022-01-20 13:07:50.012 10159-10216/com.google.android.exoplayer2.demo E/MediaCodecVideoRenderer: Video codec error
      android.media.MediaCodec$CodecException: Error 0x80000000
2022-01-20 13:07:50.037 10159-10216/com.google.android.exoplayer2.demo E/ExoPlayerImplInternal: Playback error
      com.google.android.exoplayer2.ExoPlaybackException: MediaCodecVideoRenderer error, index=0, format=Format(1, null, null, video/avc, avc1.4D0033, -1, null, [1920, 1080, 59.940063], [-1, -1]), format_supported=YES
        at com.google.android.exoplayer2.ExoPlayerImplInternal.handleMessage(ExoPlayerImplInternal.java:562)
        at android.os.Handler.dispatchMessage(Handler.java:102)
        at android.os.Looper.loopOnce(Looper.java:201)
        at android.os.Looper.loop(Looper.java:288)
        at android.os.HandlerThread.run(HandlerThread.java:67)
     Caused by: com.google.android.exoplayer2.video.MediaCodecVideoDecoderException: Decoder failed: c2.qti.avc.decoder
        at com.google.android.exoplayer2.video.MediaCodecVideoRenderer.createDecoderException(MediaCodecVideoRenderer.java:1470)
        at com.google.android.exoplayer2.mediacodec.MediaCodecRenderer.render(MediaCodecRenderer.java:799)
        at com.google.android.exoplayer2.ExoPlayerImplInternal.doSomeWork(ExoPlayerImplInternal.java:982)
        at com.google.android.exoplayer2.ExoPlayerImplInternal.handleMessage(ExoPlayerImplInternal.java:486)
        at android.os.Handler.dispatchMessage(Handler.java:102) 
        at android.os.Looper.loopOnce(Looper.java:201) 
        at android.os.Looper.loop(Looper.java:288) 
        at android.os.HandlerThread.run(HandlerThread.java:67) 
     Caused by: android.media.MediaCodec$CodecException: Error 0x80000000
2022-01-20 13:07:50.042 10159-10216/com.google.android.exoplayer2.demo E/ExoPlayerImplInternal: Disable failed.
      java.lang.IllegalStateException
        at android.media.MediaCodec.native_flush(Native Method)
        at android.media.MediaCodec.flush(MediaCodec.java:2350)
        at com.google.android.exoplayer2.mediacodec.AsynchronousMediaCodecAdapter.flush(AsynchronousMediaCodecAdapter.java:244)
        at com.google.android.exoplayer2.mediacodec.MediaCodecRenderer.flushCodec(MediaCodecRenderer.java:850)
        at com.google.android.exoplayer2.mediacodec.MediaCodecRenderer.flushOrReleaseCodec(MediaCodecRenderer.java:843)
        at com.google.android.exoplayer2.mediacodec.MediaCodecRenderer.onDisabled(MediaCodecRenderer.java:696)
        at com.google.android.exoplayer2.video.MediaCodecVideoRenderer.onDisabled(MediaCodecVideoRenderer.java:529)
        at com.google.android.exoplayer2.BaseRenderer.disable(BaseRenderer.java:178)
        at com.google.android.exoplayer2.ExoPlayerImplInternal.disableRenderer(ExoPlayerImplInternal.java:1642)
        at com.google.android.exoplayer2.ExoPlayerImplInternal.resetInternal(ExoPlayerImplInternal.java:1382)
        at com.google.android.exoplayer2.ExoPlayerImplInternal.stopInternal(ExoPlayerImplInternal.java:1345)
        at com.google.android.exoplayer2.ExoPlayerImplInternal.handleMessage(ExoPlayerImplInternal.java:578)
        at android.os.Handler.dispatchMessage(Handler.java:102)
        at android.os.Looper.loopOnce(Looper.java:201)
        at android.os.Looper.loop(Looper.java:288)
        at android.os.HandlerThread.run(HandlerThread.java:67)
2022-01-20 13:07:50.042 10159-10159/com.google.android.exoplayer2.demo D/EventLogger: videoDisabled [eventTime=7.34, mediaPos=0.00, window=0, period=0]
2022-01-20 13:07:50.046 10159-10231/com.google.android.exoplayer2.demo D/CCodecBuffers: [c2.android.aac.decoder#867:1D-Output.Impl[N]] Client returned a buffer it does not own according to our record: 0
2022-01-20 13:07:50.046 10159-10231/com.google.android.exoplayer2.demo D/CCodecBuffers: [c2.android.aac.decoder#867:1D-Output.Impl[N]] Client returned a buffer it does not own according to our record: 1
2022-01-20 13:07:50.046 10159-10231/com.google.android.exoplayer2.demo D/CCodecBuffers: [c2.android.aac.decoder#867:1D-Output.Impl[N]] Client returned a buffer it does not own according to our record: 2
2022-01-20 13:07:50.046 10159-10216/com.google.android.exoplayer2.demo D/MediaCodec: keep callback message for reclaim
2022-01-20 13:07:50.047 10159-10231/com.google.android.exoplayer2.demo I/CCodecConfig: query failed after returning 19 values (BAD_INDEX)
2022-01-20 13:07:50.047 10159-10231/com.google.android.exoplayer2.demo W/Codec2Client: query -- param skipped: index = 1342179345.
2022-01-20 13:07:50.047 10159-10231/com.google.android.exoplayer2.demo W/Codec2Client: query -- param skipped: index = 2415921170.
2022-01-20 13:07:50.047 10159-10231/com.google.android.exoplayer2.demo W/Codec2Client: query -- param skipped: index = 1610614798.
2022-01-20 13:07:50.053 10159-10231/com.google.android.exoplayer2.demo D/CCodecBufferChannel: [c2.android.aac.decoder#867] MediaCodec discarded an unknown buffer
2022-01-20 13:07:50.053 10159-10231/com.google.android.exoplayer2.demo D/CCodecBufferChannel: [c2.android.aac.decoder#867] MediaCodec discarded an unknown buffer
2022-01-20 13:07:50.054 10159-10231/com.google.android.exoplayer2.demo D/CCodecBufferChannel: [c2.android.aac.decoder#867] MediaCodec discarded an unknown buffer
2022-01-20 13:07:50.054 10159-10231/com.google.android.exoplayer2.demo D/CCodecBufferChannel: [c2.android.aac.decoder#867] MediaCodec discarded an unknown buffer
2022-01-20 13:07:50.055 10159-10231/com.google.android.exoplayer2.demo I/hw-BpHwBinder: onLastStrongRef automatically unlinking death recipients
2022-01-20 13:07:50.072 10159-10159/com.google.android.exoplayer2.demo D/EventLogger: audioDisabled [eventTime=7.37, mediaPos=0.00, window=0, period=0]
2022-01-20 13:07:50.072 10159-10159/com.google.android.exoplayer2.demo D/EventLogger: videoDecoderReleased [eventTime=7.38, mediaPos=0.00, window=0, period=0, c2.qti.avc.decoder]
2022-01-20 13:07:50.073 10159-10159/com.google.android.exoplayer2.demo D/EventLogger: audioDecoderReleased [eventTime=7.38, mediaPos=0.00, window=0, period=0, c2.android.aac.decoder]
2022-01-20 13:07:50.074 10159-10159/com.google.android.exoplayer2.demo E/EventLogger: playerFailed [eventTime=7.38, mediaPos=0.00, window=0, period=0, errorCode=ERROR_CODE_DECODING_FAILED
      com.google.android.exoplayer2.ExoPlaybackException: MediaCodecVideoRenderer error, index=0, format=Format(1, null, null, video/avc, avc1.4D0033, -1, null, [1920, 1080, 59.940063], [-1, -1]), format_supported=YES
        at com.google.android.exoplayer2.ExoPlayerImplInternal.handleMessage(ExoPlayerImplInternal.java:562)
        at android.os.Handler.dispatchMessage(Handler.java:102)
        at android.os.Looper.loopOnce(Looper.java:201)
        at android.os.Looper.loop(Looper.java:288)
        at android.os.HandlerThread.run(HandlerThread.java:67)
     Caused by: com.google.android.exoplayer2.video.MediaCodecVideoDecoderException: Decoder failed: c2.qti.avc.decoder
        at com.google.android.exoplayer2.video.MediaCodecVideoRenderer.createDecoderException(MediaCodecVideoRenderer.java:1470)
        at com.google.android.exoplayer2.mediacodec.MediaCodecRenderer.render(MediaCodecRenderer.java:799)
        at com.google.android.exoplayer2.ExoPlayerImplInternal.doSomeWork(ExoPlayerImplInternal.java:982)
        at com.google.android.exoplayer2.ExoPlayerImplInternal.handleMessage(ExoPlayerImplInternal.java:486)
        at android.os.Handler.dispatchMessage(Handler.java:102) 
        at android.os.Looper.loopOnce(Looper.java:201) 
        at android.os.Looper.loop(Looper.java:288) 
        at android.os.HandlerThread.run(HandlerThread.java:67) 
     Caused by: android.media.MediaCodec$CodecException: Error 0x80000000
    ]
2022-01-20 13:07:50.075 10159-10159/com.google.android.exoplayer2.demo D/EventLogger: state [eventTime=7.38, mediaPos=0.00, window=0, period=0, IDLE]
2022-01-20 13:07:54.873 10159-10224/com.google.android.exoplayer2.demo D/BufferPoolAccessor2.0: bufferpool2 0x7388e36e68 : 0(0 size) total buffers - 0(0 size) used buffers - 27/32 (recycle/alloc) - 6/32 (fetch/transfer)
2022-01-20 13:07:54.873 10159-10224/com.google.android.exoplayer2.demo D/BufferPoolAccessor2.0: evictor expired: 1, evicted: 1
2022-01-20 13:07:55.874 10159-10224/com.google.android.exoplayer2.demo D/BufferPoolAccessor2.0: bufferpool2 0x7388e36088 : 0(0 size) total buffers - 0(0 size) used buffers - 19/29 (recycle/alloc) - 5/24 (fetch/transfer)
2022-01-20 13:07:55.874 10159-10224/com.google.android.exoplayer2.demo D/BufferPoolAccessor2.0: evictor expired: 1, evicted: 1
```

The device I've been using to reproduce this issue has the following details:
- **ExoPlayer version number:** 2.16.1
- **Android version:** 12 (API 31)
- **Android device:** Google Pixel 5

Some further observations I've made during my investigation were that this seems to occur with the `c2.qti.avc.decoder` decoder, however when testing on a Samsung Galaxy S20 Ultra (5G), I found the issue to not occur with the `OMX.Exynos.avc.dec` decoder.

I have done a lot of searching within the issues contained in this project, as well as wider searches across Google and SO. I've seen some mildly similar issues, but nothing with a resolution.

I was able to consistently reproduce the issue with my implementation, so I went about creating an [SSCCE](http://sscce.org/) in order to rule out as many variables introduced by my full implementation, and the issue was still consistently reproduced. So as a final step, I pulled down the [ExoPlayer Demo](https://github.com/google/ExoPlayer.git) and modified the source (as per the [official documentation](https://exoplayer.dev/demo-application.html#playing-your-own-content)) and was able to reproduce the issue there also.

The `adb bugreport` output for this issue reproduction can be see here: 
[bugreport-redfin-SQ1A.220105.002-2022-01-20-13-24-56.zip](https://github.com/google/ExoPlayer/files/7901815/bugreport-redfin-SQ1A.220105.002-2022-01-20-13-24-56.zip)

I've used [MediaInfo](https://mediaarea.net/en/MediaInfo) to interrogate the video file to see if anything stood out, however I couldn't see anything amiss. I've included it below, in case it helps.

```
General
Complete name                            : C:\Users\Tom Larcher\StudioProjects\VideoPlayer\app\src\main\res\raw\example_video.MP4
Format                                   : MPEG-4
Format profile                           : JVT
Codec ID                                 : avc1 (avc1/isom)
File size                                : 20.0 MiB
Duration                                 : 6 s 523 ms
Overall bit rate mode                    : Variable
Overall bit rate                         : 25.7 Mb/s
Encoded date                             : UTC 2021-04-23 16:22:39
Tagged date                              : UTC 2021-04-23 16:22:39

Video
ID                                       : 1
Format                                   : AVC
Format/Info                              : Advanced Video Codec
Format profile                           : Main@L5.1
Format settings                          : CABAC / 1 Ref Frames
Format settings, CABAC                   : Yes
Format settings, Reference frames        : 1 frame
Format settings, GOP                     : M=1, N=60
Codec ID                                 : avc1
Codec ID/Info                            : Advanced Video Coding
Duration                                 : 6 s 523 ms
Bit rate mode                            : Variable
Bit rate                                 : 21.7 Mb/s
Maximum bit rate                         : 18.0 Mb/s
Width                                    : 1 920 pixels
Height                                   : 1 080 pixels
Display aspect ratio                     : 16:9
Frame rate mode                          : Constant
Frame rate                               : 59.940 (60000/1001) FPS
Color space                              : YUV
Chroma subsampling                       : 4:2:0
Bit depth                                : 8 bits
Scan type                                : Progressive
Bits/(Pixel*Frame)                       : 0.175
Stream size                              : 16.9 MiB (84%)
Title                                    : Ambarella AVC
Language                                 : English
Encoded date                             : UTC 2021-04-23 16:22:39
Tagged date                              : UTC 2021-04-23 16:22:39
Color range                              : Limited
Color primaries                          : BT.709
Transfer characteristics                 : BT.709
Matrix coefficients                      : BT.709
Codec configuration box                  : avcC

Audio
ID                                       : 2
Format                                   : AAC LC
Format/Info                              : Advanced Audio Codec Low Complexity
Codec ID                                 : mp4a-40-2
Duration                                 : 6 s 507 ms
Bit rate mode                            : Constant
Bit rate                                 : 128 kb/s
Nominal bit rate                         : 48.0 kb/s
Channel(s)                               : 1 channel
Channel layout                           : C
Sampling rate                            : 48.0 kHz
Frame rate                               : 46.875 FPS (1024 SPF)
Compression mode                         : Lossy
Stream size                              : 102 KiB (0%)
Title                                    : Ambarella AAC
Language                                 : English
Encoded date                             : UTC 2021-04-23 16:22:39
Tagged date                              : UTC 2021-04-23 16:22:39

Text
ID                                       : 3
Format                                   : Timed Text
Codec ID                                 : text
Duration                                 : 6 s 500 ms
Bit rate mode                            : Constant
Bit rate                                 : 10.9 kb/s
Stream size                              : 8.63 KiB (0%)
Title                                    : Ambarella EXT
Language                                 : English
Forced                                   : No
Encoded date                             : UTC 2021-04-23 16:22:39
Tagged date                              : UTC 2021-04-23 16:22:39
```

I can share any of my reproduction repos if required.

**Comment by christosts** [comment]
Thank you for sharing detailed information. Will you also be able to share a link to the mp4 file so that we test/reproduce too?

If you're unable to share it publicly, please send them to dev.exoplayer@gmail.com with subject "Issue #9879". Please also update this issue to indicate you've done this.

**Comment by findyourexit** [comment]
Thanks for picking this up @christosts! I've just sent through an email to the appropriate address with the provided subject, with the reproduction media file attached. Fortunately it's a nice small ~20mb file with little/no sensitive content. 

Please let me know if you haven't received that, and I'll host it/send through a link.

**Comment by insinfo** [comment]
I'm developing an app with Flutter and I'm using the video_player plugin which uses ExoPlayer internally, I'm also facing this issue of seeing a very large amount of log messages, this bunch of log messages get in the way of debugging my code, it would be nice if ExoPlayer removes these log messages, and leaves the log clean.

https://www.experimenteriodasostras.com.br/storage/turismo/midias/ffec8800-09d2-45a4-9b99-09ab04b057bf.mp4

```console
I/flutter (17498): -- TurismoModule INITIALIZED
I/ExoPlayerImpl(17498): Init 5797912 [ExoPlayerLib/2.14.1] [emulator64_x86_64_arm64, sdk_gphone64_x86_64, Google, 32]
D/EGL_emulation(17498): app_time_stats: avg=29338.28ms min=29338.28ms max=29338.28ms count=1
D/MetadataUtil(17498): Skipped unknown metadata entry: hdvd
D/CCodec  (17498): allocate(c2.goldfish.h264.decoder)
I/CCodec  (17498): setting up 'default' as default (vendor) store
I/CCodec  (17498): Created component [c2.goldfish.h264.decoder]
D/CCodecConfig(17498): read media type: video/avc
D/ReflectedParamUpdater(17498): extent() != 1 for single value type: algo.buffers.max-count.values
D/ReflectedParamUpdater(17498): extent() != 1 for single value type: output.subscribed-indices.values
D/ReflectedParamUpdater(17498): extent() != 1 for single value type: input.buffers.allocator-ids.values
D/ReflectedParamUpdater(17498): extent() != 1 for single value type: output.buffers.allocator-ids.values
D/ReflectedParamUpdater(17498): extent() != 1 for single value type: algo.buffers.allocator-ids.values
D/ReflectedParamUpdater(17498): extent() != 1 for single value type: output.buffers.pool-ids.values
D/ReflectedParamUpdater(17498): extent() != 1 for single value type: algo.buffers.pool-ids.values
D/ReflectedParamUpdater(17498): ignored struct field coded.color-format.locations
D/CCodecConfig(17498): ignoring local param raw.size (0xd2001800) as it is already supported
D/CCodecConfig(17498): ignoring local param default.color (0x5200180b) as it is already supported
D/ReflectedParamUpdater(17498): ignored struct field raw.hdr-static-info.mastering
I/CCodecConfig(17498): query failed after returning 12 values (BAD_INDEX)
D/CCodecConfig(17498): c2 config diff is Dict {
D/CCodecConfig(17498):   c2::u32 coded.pl.level = 20496
D/CCodecConfig(17498):   c2::u32 coded.pl.profile = 20481
D/CCodecConfig(17498):   c2::u32 coded.vui.color.matrix = 0
D/CCodecConfig(17498):   c2::u32 coded.vui.color.primaries = 0
D/CCodecConfig(17498):   c2::u32 coded.vui.color.range = 2
D/CCodecConfig(17498):   c2::u32 coded.vui.color.transfer = 0
D/CCodecConfig(17498):   c2::u32 default.color.matrix = 0
D/CCodecConfig(17498):   c2::u32 default.color.primaries = 0
D/CCodecConfig(17498):   c2::u32 default.color.range = 0
D/CCodecConfig(17498):   c2::u32 default.color.transfer = 0
D/CCodecConfig(17498):   c2::u32 input.buffers.max-size.value = 2097152
D/CCodecConfig(17498):   c2::u32 input.delay.value = 0
D/CCodecConfig(17498):   string input.media-type.value = "video/avc"
D/CCodecConfig(17498):   c2::u32 output.delay.value = 8
D/CCodecConfig(17498):   string output.media-type.value = "video/raw"
D/CCodecConfig(17498):   c2::u32 raw.color.matrix = 0
D/CCodecConfig(17498):   c2::u32 raw.color.primaries = 0
D/CCodecConfig(17498):   c2::u32 raw.color.range = 2
D/CCodecConfig(17498):   c2::u32 raw.color.transfer = 0
D/CCodecConfig(17498):   c2::u32 raw.max-size.height = 240
D/CCodecConfig(17498):   c2::u32 raw.max-size.width = 320
D/CCodecConfig(17498):   c2::u32 raw.pixel-format.value = 35
D/CCodecConfig(17498):   c2::i32 raw.rotation.flip = 0
D/CCodecConfig(17498):   c2::i32 raw.rotation.value = 0
D/CCodecConfig(17498):   c2::u32 raw.sar.height = 1
D/CCodecConfig(17498):   c2::u32 raw.sar.width = 1
D/CCodecConfig(17498):   c2::u32 raw.size.height = 240
D/CCodecConfig(17498):   c2::u32 raw.size.width = 320
D/CCodecConfig(17498):   c2:
W/ColorUtils(17498): expected specified color aspects (2:0:0:0)
D/SurfaceUtils(17498): connecting to surface 0x799202d3a950, reason connectToSurface
I/MediaCodec(17498): [c2.goldfish.h264.decoder] setting surface generation to 17917955
D/SurfaceUtils(17498): disconnecting from surface 0x799202d3a950, reason connectToSurface(reconnect)
D/SurfaceUtils(17498): connecting to surface 0x799202d3a950, reason connectToSurface(reconnect)
D/CCodec  (17498): [c2.goldfish.h264.decoder] buffers are bound to CCodec for this session
D/CCodecConfig(17498): no c2 equivalents for csd-1
D/CCodecConfig(17498): no c2 equivalents for native-window
D/CCodecConfig(17498): no c2 equivalents for flags
D/CCodecConfig(17498): config failed => CORRUPTED
D/CCodecConfig(17498): c2 config diff is   c2::u32 raw.max-size.height = 1080
D/CCodecConfig(17498):   c2::u32 raw.max-size.width = 1920
D/CCodecConfig(17498):   c2::u32 raw.size.height = 1080
D/CCodecConfig(17498):   c2::u32 raw.size.width = 1920
W/Codec2Client(17498): query -- param skipped: index = 1107298332.
D/CCodec  (17498): client requested max input size 454060, which is smaller than what component recommended (2097152); overriding with component recommendation.
W/CCodec  (17498): This behavior is subject to change. It is recommended that app developers double check whether the requested max input size is in reasonable range.
D/CCodec  (17498): setup formats input: AMessage(what = 0x00000000) = {
D/CCodec  (17498):   int32_t height = 1080
D/CCodec  (17498):   int32_t level = 65536
D/CCodec  (17498):   int32_t max-input-size = 2097152
D/CCodec  (17498):   string mime = "video/avc"
D/CCodec  (17498):   int32_t profile = 65536
D/CCodec  (17498):   int32_t width = 1920
D/CCodec  (17498):   Rect crop(0, 0, 1919, 1079)
D/CCodec  (17498): }
D/CCodec  (17498): setup formats output: AMessage(what = 0x00000000) = {
D/CCodec  (17498):   int32_t android._color-format = 2135033992
D/CCodec  (17498):   int32_t android._video-scaling = 1
D/CCodec  (17498):   int32_t rotation-degrees = 0
D/CCodec  (17498):   int32_t color-standard = 1
D/CCodec  (17498):   int32_t color-range = 2
D/CCodec  (17498):   int32_t color-transfer = 3
D/CCodec  (17498):   int32_t sar-height = 1
D/CCodec  (17498):   int32_t sar-width = 1
D/CCodec  (17498):   Rect crop(0, 0, 1919, 1079)
D/CCodec  (17498):   int32_t width = 1920
D/CCodec  (17498):   int32_t height = 1080
D/CCodec  (17498):   int32_t max-height = 1080
D/CCodec  (17498):   int32_t max-width = 1920
D/CCodec  (17498):   string mime = "video/raw"
D/CCodec  (17498):   int32_t android._dataspace = 260
D/CCodec  (17498):   int32_t color-format = 2130708361
D/CCodec  (17498): }
I/CCodecConfig(17498): query failed after returning 12 values (BAD_INDEX)
W/Codec2Client(17498): query -- param skipped: index = 1342179345.
W/Codec2Client(17498): query -- param skipped: index = 2415921170.
W/Codec2Client(17498): query -- param skipped: index = 1610614798.
W/Codec2Client(17498): query -- param skipped: index = 2684356609.
D/CCodecBufferChannel(17498): [c2.goldfish.h264.decoder#492] Created input block pool with allocatorID 16 => poolID 18 - OK (0)
D/CCodecBufferChannel(17498): [c2.goldfish.h264.decoder#492] Query output surface allocator returned 0 params => BAD_INDEX (6)
I/CCodecBufferChannel(17498): [c2.goldfish.h264.decoder#492] Created output block pool with allocatorID 18 => poolID 18 - OK
D/CCodecBufferChannel(17498): [c2.goldfish.h264.decoder#492] Configured output block pool ids 18 => OK
D/Codec2-OutputBufferQueue(17498): remote graphic buffer migration 0/0
D/Codec2Client(17498): setOutputSurface -- failed to set consumer usage (6/BAD_INDEX)
D/Codec2Client(17498): setOutputSurface -- generation=17917955 consumer usage=0x900
D/Codec2Client(17498): Surface configure completed
D/BufferPoolAccessor2.0(17498): bufferpool2 0x799162ce9d68 : 0(0 size) total buffers - 0(0 size) used buffers - 0/4 (recycle/alloc) - 4/5277 (fetch/transfer)
D/BufferPoolAccessor2.0(17498): Destruction - bufferpool2 0x799162ce9d68 cached: 0/0M, 0/0% in use; allocs: 4, 0% recycled; transfers: 5277, 100% unfetched
D/CCodecConfig(17498): c2 config diff is   c2::u32 raw.crop.height = 1080
D/CCodecConfig(17498):   c2::u32 raw.crop.left = 0
D/CCodecConfig(17498):   c2::u32 raw.crop.top = 0
D/CCodecConfig(17498):   c2::u32 raw.crop.width = 1920
W/MediaCodec(17498): mapFormat: no mediaType information
I/flutter (17498): initState true
D/EGL_emulation(17498): app_time_stats: avg=457.52ms min=170.30ms max=1086.75ms count=4
D/CCodecBuffers(17498): [c2.goldfish.h264.decoder#492:1D-Input.Impl[N]] codec released a buffer owned by client (index 1)
D/EGL_emulation(17498): app_time_stats: avg=69.47ms min=13.19ms max=182.53ms count=15
D/CCodecBuffers(17498): [c2.goldfish.h264.decoder#492:1D-Input.Impl[N]] codec released a buffer owned by client (index 0)
I/riodasostrasap(17498): Background young concurrent copying GC freed 18107(1697KB) AllocSpace objects, 5(260KB) LOS objects, 3% free, 64MB/66MB, paused 660us,79us total 106.813ms
D/EGL_emulation(17498): app_time_stats: avg=52.24ms min=10.82ms max=146.39ms count=20
D/CCodecBuffers(17498): [c2.goldfish.h264.decoder#492:1D-Input.Impl[N]] codec released a buffer owned by client (index 0)
D/EGL_emulation(17498): app_time_stats: avg=67.08ms min=19.56ms max=227.04ms count=15
I/flutter (17498): -- TurismoModule DISPOSED
I/riodasostrasap(17498): Background young concurrent copying GC freed 1636(142KB) AllocSpace objects, 0(0B) LOS objects, 0% free, 66MB/66MB, paused 469us,85us total 144.306ms
D/BufferPoolAccessor2.0(17498): bufferpool2 0x799162dbf058 : 4(8388608 size) total buffers - 4(8388608 size) used buffers - 0/4 (recycle/alloc) - 4/121 (fetch/transfer)
D/BufferPoolAccessor2.0(17498): evictor expired: 1, evicted: 1
D/CCodecBuffers(17498): [c2.goldfish.h264.decoder#492:1D-Input.Impl[N]] codec released a buffer owned by client (index 1)
D/CCodecBuffers(17498): [c2.goldfish.h264.decoder#492:1D-Input.Impl[N]] codec released a buffer owned by client (index 3)
I/ExoPlayerImpl(17498): Release 5797912 [ExoPlayerLib/2.14.1] [emulator64_x86_64_arm64, sdk_gphone64_x86_64, Google, 32] [goog.exo.core]
E/BufferQueueProducer(17498): [SurfaceTexture-0-17498-1](id:445a00000004,api:3,p:17498,c:17498) dequeueBuffer: BufferQueue has been abandoned
D/CCodecBufferChannel(17498): [c2.goldfish.h264.decoder#492] work failed to complete: 14
D/CCodecBuffers(17498): [c2.goldfish.h264.decoder#492:1D-Input.Impl[N]] codec released a buffer owned by client (index 2)
E/MediaCodec(17498): Codec reported err 0xe, actionCode 0, while in state 6/STARTED
D/SurfaceUtils(17498): disconnecting from surface 0x799202d3a950, reason disconnectFromSurface
E/BufferQueueProducer(17498): [SurfaceTexture-0-17498-1](id:445a00000004,api:3,p:17498,c:17498) cancelBuffer: BufferQueue has been abandoned
E/BufferQueueProducer(17498): [SurfaceTexture-0-17498-1](id:445a00000004,api:3,p:17498,c:17498) cancelBuffer: BufferQueue has been abandoned
E/BufferQueueProducer(17498): [SurfaceTexture-0-17498-1](id:445a00000004,api:3,p:17498,c:17498) cancelBuffer: BufferQueue has been abandoned
E/BufferQueueProducer(17498): [SurfaceTexture-0-17498-1](id:445a00000004,api:3,p:17498,c:17498) cancelBuffer: BufferQueue has been abandoned
E/BufferQueueProducer(17498): [SurfaceTexture-0-17498-1](id:445a00000004,api:3,p:17498,c:17498) cancelBuffer: BufferQueue has been abandoned
E/BufferQueueProducer(17498): [SurfaceTexture-0-17498-1](id:445a00000004,api:3,p:17498,c:17498) cancelBuffer: BufferQueue has been abandoned
E/BufferQueueProducer(17498): [SurfaceTexture-0-17498-1](id:445a00000004,api:3,p:17498,c:17498) cancelBuffer: BufferQueue has been abandoned
E/BufferQueueProducer(17498): [SurfaceTexture-0-17498-1](id:445a00000004,api:3,p:17498,c:17498) cancelBuffer: BufferQueue has been abandoned
E/BufferQueueProducer(17498): [SurfaceTexture-0-17498-1](id:445a00000004,api:3,p:17498,c:17498) cancelBuffer: BufferQueue has been abandoned
E/BufferQueueProducer(17498): [SurfaceTexture-0-17498-1](id:445a00000004,api:3,p:17498,c:17498) cancelBuffer: BufferQueue has been abandoned
E/BufferQueueProducer(17498): [SurfaceTexture-0-17498-1](id:445a00000004,api:3,p:17498,c:17498) cancelBuffer: BufferQueue has been abandoned
E/BufferQueueProducer(17498): [SurfaceTexture-0-17498-1](id:445a00000004,api:3,p:17498,c:17498) cancelBuffer: BufferQueue has been abandoned
E/MediaCodecVideoRenderer(17498): Video codec error
E/MediaCodecVideoRenderer(17498):   java.lang.IllegalStateException
E/MediaCodecVideoRenderer(17498):       at android.media.MediaCodec.native_dequeueInputBuffer(Native Method)
E/MediaCodecVideoRenderer(17498):       at android.media.MediaCodec.dequeueInputBuffer(MediaCodec.java:2938)
E/MediaCodecVideoRenderer(17498):       at com.google.android.exoplayer2.mediacodec.SynchronousMediaCodecAdapter.dequeueInputBufferIndex(SynchronousMediaCodecAdapter.java:93)
E/MediaCodecVideoRenderer(17498):       at com.google.android.exoplayer2.mediacodec.MediaCodecRenderer.feedInputBuffer(MediaCodecRenderer.java:1209)
E/MediaCodecVideoRenderer(17498):       at com.google.android.exoplayer2.mediacodec.MediaCodecRenderer.render(MediaCodecRenderer.java:845)
E/MediaCodecVideoRenderer(17498):       at com.google.android.exoplayer2.ExoPlayerImplInternal.doSomeWork(ExoPlayerImplInternal.java:945)
E/MediaCodecVideoRenderer(17498):       at com.google.android.exoplayer2.ExoPlayerImplInternal.handleMessage(ExoPlayerImplInternal.java:478)
E/MediaCodecVideoRenderer(17498):       at android.os.Handler.dispatchMessage(Handler.java:102)
E/MediaCodecVideoRenderer(17498):       at android.os.Looper.loopOnce(Looper.java:201)
E/MediaCodecVideoRenderer(17498):       at android.os.Looper.loop(Looper.java:288)
E/MediaCodecVideoRenderer(17498):       at android.os.HandlerThread.run(HandlerThread.java:67)
E/ExoPlayerImplInternal(17498): Playback error
E/ExoPlayerImplInternal(17498):   com.google.android.exoplayer2.ExoPlaybackException: MediaCodecVideoRenderer error, index=0, format=Format(1, null, null, video/avc, avc1.640028, -1, null, [1920, 1080, 29.970032], [-1, -1]), format_supported=YES
E/ExoPlayerImplInternal(17498):       at com.google.android.exoplayer2.ExoPlayerImplInternal.handleMessage(ExoPlayerImplInternal.java:555)
E/ExoPlayerImplInternal(17498):       at android.os.Handler.dispatchMessage(Handler.java:102)
E/ExoPlayerImplInternal(17498):       at android.os.Looper.loopOnce(Looper.java:201)
E/ExoPlayerImplInternal(17498):       at android.os.Looper.loop(Looper.java:288)
E/ExoPlayerImplInternal(17498):       at android.os.HandlerThread.run(HandlerThread.java:67)
E/ExoPlayerImplInternal(17498):   Caused by: com.google.android.exoplayer2.video.MediaCodecVideoDecoderException: Decoder failed: c2.goldfish.h264.decoder
E/ExoPlayerImplInternal(17498):       at com.google.android.exoplayer2.video.MediaCodecVideoRenderer.createDecoderException(MediaCodecVideoRenderer.java:1410)
E/ExoPlayerImplInternal(17498):       at com.google.android.exoplayer2.mediacodec.MediaCodecRenderer.render(MediaCodecRenderer.java:864)
E/ExoPlayerImplInternal(17498):       at com.google.android.exoplayer2.ExoPlayerImplInternal.doSomeWork(ExoPlayerImplInternal.java:945)
E/ExoPlayerImplInternal(17498):       at com.google.android.exoplayer2.ExoPlayerImplInternal.handleMessage(ExoPlayerImplInternal.java:478)
E/ExoPlayerImplInternal(17498):       ... 4 more
E/ExoPlayerImplInternal(17498):   Caused by: java.lang.IllegalStateException
E/ExoPlayerImplInternal(17498):       at android.media.MediaCodec.native_dequeueInputBuffer(Native Method)
E/ExoPlayerImplInternal(17498):       at android.media.MediaCodec.dequeueInputBuffer(MediaCodec.java:2938)
E/ExoPlayerImplInternal(17498):       at com.google.android.exoplayer2.mediacodec.SynchronousMediaCodecAdapter.dequeueInputBufferIndex(SynchronousMediaCodecAdapter.java:93)
E/ExoPlayerImplInternal(17498):       at com.google.android.exoplayer2.mediacodec.MediaCodecRenderer.feedInputBuffer(MediaCodecRenderer.java:1209)
E/ExoPlayerImplInternal(17498):       at com.google.android.exoplayer2.mediacodec.MediaCodecRenderer.render(MediaCodecRenderer.java:845)
E/ExoPlayerImplInternal(17498):       ... 6 more
E/VideoFrameReleaseHelper(17498): Failed to call Surface.setFrameRate
E/VideoFrameReleaseHelper(17498):   java.lang.IllegalStateException: Surface has already been released.
E/VideoFrameReleaseHelper(17498):       at android.view.Surface.checkNotReleasedLocked(Surface.java:772)
E/VideoFrameReleaseHelper(17498):       at android.view.Surface.setFrameRate(Surface.java:978)
E/VideoFrameReleaseHelper(17498):       at android.view.Surface.setFrameRate(Surface.java:997)
E/VideoFrameReleaseHelper(17498):       at com.google.android.exoplayer2.video.VideoFrameReleaseHelper.setSurfaceFrameRateV30(VideoFrameReleaseHelper.java:389)
E/VideoFrameReleaseHelper(17498):       at com.google.android.exoplayer2.video.VideoFrameReleaseHelper.clearSurfaceFrameRate(VideoFrameReleaseHelper.java:379)
E/VideoFrameReleaseHelper(17498):       at com.google.android.exoplayer2.video.VideoFrameReleaseHelper.onStopped(VideoFrameReleaseHelper.java:232)
E/VideoFrameReleaseHelper(17498):       at com.google.android.exoplayer2.video.MediaCodecVideoRenderer.onStopped(MediaCodecVideoRenderer.java:471)
E/VideoFrameReleaseHelper(17498):       at com.google.android.exoplayer2.BaseRenderer.stop(BaseRenderer.java:165)
E/VideoFrameReleaseHelper(17498):       at com.google.android.exoplayer2.ExoPlayerImplInternal.ensureStopped(ExoPlayerImplInternal.java:1599)
E/VideoFrameReleaseHelper(17498):       at com.google.android.exoplayer2.ExoPlayerImplInternal.disableRenderer(ExoPlayerImplInternal.java:1608)
E/VideoFrameReleaseHelper(17498):       at com.google.android.exoplayer2.ExoPlayerImplInternal.resetInternal(ExoPlayerImplInternal.java:1351)
E/VideoFrameReleaseHelper(17498):       at com.google.android.exoplayer2.ExoPlayerImplInternal.stopInternal(ExoPlayerImplInternal.java:1314)
E/VideoFrameReleaseHelper(17498):       at com.google.android.exoplayer2.ExoPlayerImplInternal.handleMessage(ExoPlayerImplInternal.java:571)
E/VideoFrameReleaseHelper(17498):       at android.os.Handler.dispatchMessage(Handler.java:102)
E/VideoFrameReleaseHelper(17498):       at android.os.Looper.loopOnce(Looper.java:201)
E/VideoFrameReleaseHelper(17498):       at android.os.Looper.loop(Looper.java:288)
E/VideoFrameReleaseHelper(17498):       at android.os.HandlerThread.run(HandlerThread.java:67)
E/ExoPlayerImplInternal(17498): Disable failed.
E/ExoPlayerImplInternal(17498):   java.lang.IllegalStateException
E/ExoPlayerImplInternal(17498):       at android.media.MediaCodec.native_flush(Native Method)
E/ExoPlayerImplInternal(17498):       at android.media.MediaCodec.flush(MediaCodec.java:2350)
E/ExoPlayerImplInternal(17498):       at com.google.android.exoplayer2.mediacodec.SynchronousMediaCodecAdapter.flush(SynchronousMediaCodecAdapter.java:160)
E/ExoPlayerImplInternal(17498):       at com.google.android.exoplayer2.mediacodec.MediaCodecRenderer.flushCodec(MediaCodecRenderer.java:912)
E/ExoPlayerImplInternal(17498):       at com.google.android.exoplayer2.mediacodec.MediaCodecRenderer.flushOrReleaseCodec(MediaCodecRenderer.java:905)
E/ExoPlayerImplInternal(17498):       at com.google.android.exoplayer2.mediacodec.MediaCodecRenderer.onDisabled(MediaCodecRenderer.java:761)
E/ExoPlayerImplInternal(17498):       at com.google.android.exoplayer2.video.MediaCodecVideoRenderer.onDisabled(MediaCodecVideoRenderer.java:483)
E/ExoPlayerImplInternal(17498):       at com.google.android.exoplayer2.BaseRenderer.disable(BaseRenderer.java:176)
E/ExoPlayerImplInternal(17498):       at com.google.android.exoplayer2.ExoPlayerImplInternal.disableRenderer(ExoPlayerImplInternal.java:1609)
E/ExoPlayerImplInternal(17498):       at com.google.android.exoplayer2.ExoPlayerImplInternal.resetInternal(ExoPlayerImplInternal.java:1351)
E/ExoPlayerImplInternal(17498):       at com.google.android.exoplayer2.ExoPlayerImplInternal.stopInternal(ExoPlayerImplInternal.java:1314)
E/ExoPlayerImplInternal(17498):       at com.google.android.exoplayer2.ExoPlayerImplInternal.handleMessage(ExoPlayerImplInternal.java:571)
E/ExoPlayerImplInternal(17498):       at android.os.Handler.dispatchMessage(Handler.java:102)
E/ExoPlayerImplInternal(17498):       at android.os.Looper.loopOnce(Looper.java:201)
E/ExoPlayerImplInternal(17498):       at android.os.Looper.loop(Looper.java:288)
E/ExoPlayerImplInternal(17498):       at android.os.HandlerThread.run(HandlerThread.java:67)
D/CCodecBufferChannel(17498): [c2.goldfish.h264.decoder#492] MediaCodec discarded an unknown buffer
D/CCodecBufferChannel(17498): [c2.goldfish.h264.decoder#492] MediaCodec discarded an unknown buffer
D/CCodecBufferChannel(17498): [c2.goldfish.h264.decoder#492] MediaCodec discarded an unknown buffer
D/CCodecBufferChannel(17498): [c2.goldfish.h264.decoder#492] MediaCodec discarded an unknown buffer
E/BufferQueueProducer(17498): [SurfaceTexture-0-17498-1](id:445a00000004,api:3,p:17498,c:17498) cancelBuffer: BufferQueue has been abandoned
I/hw-BpHwBinder(17498): onLastStrongRef automatically unlinking death recipients
E/BufferQueueProducer(17498): [SurfaceTexture-0-17498-1](id:445a00000004,api:3,p:17498,c:17498) detachBuffer: BufferQueue has been abandoned
I/riodasostrasap(17498): Background concurrent copying GC freed 5254(770KB) AllocSpace objects, 847(54MB) LOS objects, 49% free, 11MB/23MB, paused 1.016ms,76us total 632.646ms
D/BufferPoolAccessor2.0(17498): evictor expired: 1, evicted: 0
D/BufferPoolAccessor2.0(17498): bufferpool2 0x799162dbf058 : 0(0 size) total buffers - 0(0 size) used buffers - 0/4 (recycle/alloc) - 4/137 (fetch/transfer)
D/BufferPoolAccessor2.0(17498): evictor expired: 1, evicted: 1

```

**Comment by christosts** [comment]
@findyourexit does this happen in your app only or where you able to reproduce with the [ExoPlayer demo app](https://exoplayer.dev/demo-application.html) too? In the attached bug-report, I see logs from your app and also from the `EventLogger`, a utility listener that logs player activity used by default in the demo app, but I think you may have attached an `EventLogger` in your app too to get debug logs.

If you haven't repro'd on the demo app, can you please try so? Download the sources and change the app to not start playback automatically - change [this line](https://github.com/google/ExoPlayer/blob/r2.16.1/demos/main/src/main/java/com/google/android/exoplayer2/demo/PlayerActivity.java#L281) to call `player.setPlayWhenReady(false)`. Please let us know the outcome.

If it happens on your app alone, can you describe what your app is doing in the scenario you described? In the logs, I see the player has reached the READY state and then MediaCodec throws an exception but I cannot clearly see which MediaCodec method threw. Can you please describe how you initialize the player and if your app is interacting with the player while waiting for the user to start playback? 

Thanks

**Comment by christosts** [comment]
@insinfo your stack trace shows a different error, please open a new issue to avoid having multiple discussions that might create confusion. Please consider providing the information described in the [issue template](https://github.com/google/ExoPlayer/blob/release-v2/.github/ISSUE_TEMPLATE/bug.md). 

I see from the logs Flutter is using ExoPlayer 2.14.1 and the stack trace is taken from an emulator. May I suggest playing the video on a device with the [ExoPlayer demo app](https://exoplayer.dev/demo-application.html) too? Also, the logs you're seeing are produced from the Android platform and ExoPlayer cannot disable them.

**Comment by findyourexit** [comment]
Hi @christosts, yes I've managed to reproduce the issue in my own app, as well as a second app I created for the sole purpose of reproduction of this issue, and finally in the demo app (below quote from my original post).

> I was able to consistently reproduce the issue with my implementation, so I went about creating an [SSCCE](http://sscce.org/) in order to rule out as many variables introduced by my full implementation, and the issue was still consistently reproduced. So as a final step, I pulled down the [ExoPlayer Demo](https://github.com/google/ExoPlayer.git) and modified the source (as per the [official documentation](https://exoplayer.dev/demo-application.html#playing-your-own-content)) and was able to reproduce the issue there also.

The only changes I needed to make in order to reproduce the issue with the demo app was to add my `example_video.MP4` file to the `assets` folder, replace [this line](https://github.com/google/ExoPlayer/blob/r2.16.1/demos/main/src/main/java/com/google/android/exoplayer2/demo/PlayerActivity.java#L281) in `PlayerActivity` with `player.setPlayWhenReady(false);`, and replace the contents of `media.exolist.json` with the following:

```json
[
  {
    "name": "Sample videos",
    "samples": [
      {
        "name": "Decoder failure",
        "uri": "file:///android_asset/example_video.MP4"
      }
    ]
  }
]
```

I hope that clears things up!

**Comment by christosts** [comment]
Thanks, sorry for missing this information from the first post.

**Comment by findyourexit** [comment]
> Thanks, sorry for missing this information from the first post.

Hey no stress @christosts! It was a bit of behemoth of a post 😅

**Comment by christosts** [comment]
This appears to be a duplicate of #9256 for which we are waiting on a fix. I will close this issue as a dup of #9256. Please follow updates on the other issue which will update as soon as we have more news.
